### DISCUSSION TITLE
Treeshaking Support

### DISCUSSION THREAD

**Discussion by Matt-Jensen** [handsontable/hyperformula#1003] [Feature requests]
### Description
First of all: I'm a big fan of both [Formula parser](https://github.com/handsontable/formula-parser) as well as this project.  Kudos to everyone that made this possible!

That said, the only real detractor for using Hyperformula is overall package size.  Formula-parser is significantly smaller and fits my needs just fine.  For that reason I was wondering if it is currently possible to only import modules used?

For example instead of importing the entire project:
`import { HyperFormula } from 'hyperformula'`;

I'd like to be able to import the most minimal parser possible:
`import Parser from 'hyperformula/parser;`

Apologies if this is already supported, in which case I'd be happy to reframe this into a documentation request. 😄

**Comment by sequba** [comment]
Hi @Matt-Jensen, Thank you for your feedback. We're glad HyperFormula fits your needs.

Being treeshaking-friendly is a very good point. To be honest we didn't consider it while building HyperFormula. Typical spreadsheet applications make use of almost all the features of HyperFormula so they won't benefit from treeshaking that much, but for other use-cases it might make perfect sense.

Could you describe your application and the way you use HyperFormula/formula-parser? Do you only parse formulas without calculating them? Do your formulas contain references to other spreadsheet cells? The context of your use-case will help us understand what modules should be exposed for importing separately.

**Reply by Matt-Jensen** [reply]
Thank you for your response! I get your point about most users needing the majority of the features here, so I do appreciate you entertaining this request.

I'm building 3rd party embedded widgets so minimizing size is a key concern.  As with hot-formula-parser I'd expect to be managing state myself (to minimize package size). Even if it means all I can do is pass a single function to the parser.  So keeping functionality as minimal as:

```tsx
import parser from 'hyperformula/parser;

parser('=SUM(1, 2)'); // outputs number 3
```

So minimal needs here seem to be parsing and calculation.  No need for notion of spreadsheets, references to other cells, or even custom functions.  Does that answer your question?

**Reply by sequba** [reply]
Yes, thanks a lot. We will discuss your request on our team meeting.

**Reply by warpech** [reply]
We should analyze what elements exactly make `hyperformula` larger than `hot-formula-parser`.

<img width="2262" alt="Screen Shot 2022-06-08 at 2 52 20 PM" src="https://user-images.githubusercontent.com/566463/172621129-850ffdee-a339-42ba-a8ab-34aac8b85068.png">

<img width="2107" alt="Screen Shot 2022-06-08 at 2 57 37 PM" src="https://user-images.githubusercontent.com/566463/172622124-0f75a02c-31b5-4820-87eb-b3b6dcd2760f.png">

Source:

- https://bundlephobia.com/package/hyperformula@2.0.0
- https://bundlephobia.com/package/hot-formula-parser@4.0.0

**Comment by sequba** [comment]
Issue to deal with this problem: #1006
